File: sherpa-onnx/python/csrc/offline-medasr-ctc-model-config.cc
// sherpa-onnx/python/csrc/offline-medasr-ctc-model-config.cc
//
// Copyright (c)  2023  Xiaomi Corporation

#include "sherpa-onnx/csrc/offline-medasr-ctc-model-config.h"

#include <string>
#include <vector>

#include "sherpa-onnx/python/csrc/offline-medasr-ctc-model-config.h"

namespace sherpa_onnx {

void PybindOfflineMedAsrCtcModelConfig(py::module *m) {
  using PyClass = OfflineMedAsrCtcModelConfig;
  py::class_<PyClass>(*m, "OfflineMedAsrCtcModelConfig")
      .def(py::init<const std::string &>(), py::arg("model"))
      .def_readwrite("model", &PyClass::model)
      .def("__str__", &PyClass::ToString);
}

}  // namespace sherpa_onnx

File: sherpa-onnx/java-api/src/main/java/com/k2fsa/sherpa/onnx/OfflineMedAsrCtcModelConfig.java
package com.k2fsa.sherpa.onnx;

public class OfflineMedAsrCtcModelConfig {
    private final String model;

    private OfflineMedAsrCtcModelConfig(Builder builder) {
        this.model = builder.model;
    }

    public static Builder builder() {
        return new Builder();
    }

    public String getModel() {
        return model;
    }

    public static class Builder {
        private String model = "";

        public OfflineMedAsrCtcModelConfig build() {
            return new OfflineMedAsrCtcModelConfig(this);
        }

        public Builder setModel(String model) {
            this.model = model;
            return this;
        }
    }
}
